components/arm/fake/fake.go

```diff
@@ -0,0 +1,194 @@
+// Package fake implements a fake arm.
+package fake
+
+import (
+	"context"
+	// for arm model.
+	_ "embed"
+
+	"github.com/edaniels/golog"
+	"github.com/pkg/errors"
+	commonpb "go.viam.com/api/common/v1"
+	pb "go.viam.com/api/component/arm/v1"
+
+	"go.viam.com/rdk/components/arm"
+	"go.viam.com/rdk/components/arm/eva"
+	ur "go.viam.com/rdk/components/arm/universalrobots"
+	"go.viam.com/rdk/components/arm/xarm"
+	"go.viam.com/rdk/components/arm/yahboom"
+	"go.viam.com/rdk/components/generic"
+	"go.viam.com/rdk/config"
+	"go.viam.com/rdk/motionplan"
+	"go.viam.com/rdk/referenceframe"
+	"go.viam.com/rdk/registry"
+)
+
+// ModelName is the string used to refer to the fake arm model.
+const ModelName = "fake"
+
+//go:embed fake_model.json
+var fakeModelJSON []byte
+
+// AttrConfig is used for converting config attributes.
+type AttrConfig struct {
+	FailNew  bool   `json:"fail_new"`
+	ArmModel string `json:"arm-model"`
+}
+
+// Validate ensures all parts of the config are valid.
+func (config *AttrConfig) Validate(path string) error {
+	if config.FailNew {
+		return errors.New("whoops")
+	}
+	return nil
+}
+
+func init() {
+	registry.RegisterComponent(arm.Subtype, ModelName, registry.Component{
+		Constructor: func(ctx context.Context, _ registry.Dependencies, config config.Component, logger golog.Logger) (interface{}, error) {
+			return NewArm(config, logger)
+		},
+	})
+
+	config.RegisterComponentAttributeMapConverter(
+		arm.SubtypeName,
+		ModelName,
+		func(attributes config.AttributeMap) (interface{}, error) {
+			var conf AttrConfig
+			return config.TransformAttributeMapToStruct(&conf, attributes)
+		},
+		&AttrConfig{},
+	)
+}
+
+// NewArm returns a new fake arm.
+func NewArm(cfg config.Component, logger golog.Logger) (arm.LocalArm, error) {
+	var model referenceframe.Model
+	var err error
+	if cfg.ConvertedAttributes != nil {
+		switch cfg.ConvertedAttributes.(*AttrConfig).ArmModel {
+		case xarm.ModelName6DOF:
+			model, err = xarm.Model(6, cfg.Name)
+		case xarm.ModelName7DOF:
+			model, err = xarm.Model(7, cfg.Name)
+		case ur.ModelName:
+			model, err = ur.Model(cfg.Name)
+		case yahboom.ModelName:
+			model, err = yahboom.Model(cfg.Name)
+		case eva.ModelName:
+			model, err = eva.Model(cfg.Name)
+		case ModelName, "":
+			model, err = referenceframe.UnmarshalModelJSON(fakeModelJSON, cfg.Name)
+		default:
+			return nil, errors.Errorf("fake arm cannot be created, unsupported arm_model: %s", cfg.ConvertedAttributes.(*AttrConfig).ArmModel)
+		}
+	} else {
+		model, err = referenceframe.UnmarshalModelJSON(fakeModelJSON, cfg.Name)
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	mp, err := motionplan.NewCBiRRTMotionPlanner(model, 4, logger)
```

Comment:
Can we take this opportunity to `const` the default core count for planning?

Reply:
Yeah I'll do this